### PR TITLE
 Bytes.writeSkip should be able to accept -ve numbers #226

### PR DESCRIPTION
**Pull request by JerryShea**

### CODE REVIEW

File: src/test/java/net/openhft/chronicle/bytes/BytesTest.java [hunk 1]
```diff
@@ -45,6 +45,7 @@
 import static org.junit.Assert.*;
 import static org.junit.Assume.assumeFalse;
 import static org.junit.Assume.assumeTrue;
+import static org.junit.jupiter.api.Assertions.assertThrows;
```
Comment (JerryShea) [inline src/test/java/net/openhft/chronicle/bytes/BytesTest.java]: I did not convert the test to JUnit 5 - just did the smallest thing that could work

**Review by nicktindall — APPROVED** [review]

File: src/main/java/net/openhft/chronicle/bytes/internal/ChunkedMappedBytes.java [hunk 2]
```diff
@@ -256,10 +256,10 @@ protected void writeCheckOffset(final long offset, final long adding)
             throws BufferOverflowException, IllegalStateException {
 
         throwExceptionIfClosed();
-        if (offset < 0 || offset > mappedFile.capacity() - adding)
+        if (offset + adding < start() || offset > mappedFile.capacity() - adding)
```
Comment (nicktindall) [inline src/main/java/net/openhft/chronicle/bytes/internal/ChunkedMappedBytes.java:259]: was this a bug to be using `0` instead of `start()` previously?

Comment (JerryShea) [reply src/main/java/net/openhft/chronicle/bytes/internal/ChunkedMappedBytes.java:259]: Yes I think so

File: src/test/java/net/openhft/chronicle/bytes/BytesTest.java [hunk 3]
```diff
@@ -1049,4 +1049,20 @@ public void testReadWithOffset() {
...
+            String hello = "hello";
+            a.append(hello);
+            assertEquals(hello, a.toString());
+            a.writeSkip(-hello.length());
+            assertEquals("", a.toString());
+            if (!a.unchecked())
+                assertThrows(BufferOverflowException.class, () -> a.writeSkip(-1));
+        } finally {
+            postTest(a);
+        }
```
Comment (minborg) [inline src/test/java/net/openhft/chronicle/bytes/BytesTest.java:1066]: Nice with a test, preventing any accidental requireNonNegative() to sneak in...

**Review by minborg — APPROVED** [review]

**Pull request merged**